glfw/src/time.zig

```diff
@@ -53,12 +53,18 @@ pub inline fn getTime() f64 {
 /// base time is not atomic, so it needs to be externally synchronized with calls to glfw.getTime.
 ///
 /// see also: time
-pub inline fn setTime(time: f64) Error!void {
+pub inline fn setTime(time: f64) void {
     internal_debug.assertInitialized();
+
+    std.debug.assert(!std.math.isNan(time));
+    std.debug.assert(time >= 0);
+    // TODO: Look into why GLFW uses this hardcoded float literal as the maximum valid value for 'time'.
+    // Maybe propose upstream to name this constant. 
+    std.debug.assert(time <= 18446744073.0);
```

Comment:
Very odd, no clue myself.

Reply:
Too bad, was hoping you maybe had an inkling.

Reply:
I told my brother @Andoryuuta about this, he found the answer for us :)

The initial digits match up with 2^64, which is true:

```
18446744073.0                # GLFW constant
18446744073709551616 # 2^64
```

But why don't the remainder? Well, 1 second = 1000000000 nanoseconds, so if we multiply the constant times nanoseconds we get:

```
18446744073.0 * 1000000000  == 18446744073000000000
```

Now if we compare we see:

```
18446744073000000000 # GLFW constant in nanoseconds
18446744073709551616  # 2^64
```

And so it becomes clear: `18446744073000000000` (GLFW constant in nanoseconds) is the largest second representable by a u64 type with nanosecond precision (if we add another `1000000000`, we'd overflow)

Reply:
Huh, fancy that. Well we should probably add a comment about that in case anyone else asks a similar question in the future.

Reply:
Probably it was easier to hard-code that way due to C always being confusing :) but if one can rely on division and integer truncation, this would be a much clearer way to write it I think (Python):

```
((2**64) / (1000*1000*1000)) # largest number of seconds that can fit in uint64 with nanosecond precision
```

I'll leave the Zig port to you :)

Reply:
Went ahead and made a quick PR for it.